client/helpers.go

```diff
@@ -149,3 +151,38 @@ func accountObfusactor(aa []Account, msg string) string {
 	}
 	return msg
 }
+
+type AWSService string
+
+const (
+	ApigatewayService AWSService = "apigateway"
+)
+
+type serviceInfo struct {
+	useAccoundID bool
+}
+
+var knownServices = map[AWSService]serviceInfo{}
+
+// ResolveARN returns a column resolver that will set a field value to a proper ARN
+// based on provided AWS service and resource id value returned by resourceID function.
+func ResolveARN(service AWSService, resourceID func(resource *schema.Resource) ([]string, error)) schema.ColumnResolver {
+	return func(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+		cl := meta.(*Client)
+		idParts, err := resourceID(resource)
+		if err != nil {
+			return fmt.Errorf("error resolving resource id: %w", err)
+		}
+		var accountID string
+		if knownServices[service].useAccoundID {
+			accountID = cl.AccountID
+		}
+		return resource.Set(c.Name, arn.ARN{
+			Partition: "aws",
+			Service:   string(service),
+			Region:    cl.Region,
+			AccountID: accountID,
+			Resource:  strings.Join(append([]string{""}, idParts...), "/"),
+		}.String())
+	}
+}
```

Comment:
Is this new function completely necessary can some of the functionality from  `client.GenerateResourceARN` be used?

Reply:
This changes a bit in follow up PR, and I think I will remove `client.GenerateResourceARN` completely at the end.